### ISSUE TITLE
Support boolean options in `searchParams`?

### ISSUE THREAD

**Issue by achingbrain** [sindresorhus/ky#182]
The following code causes `ky` to throw an error:

```javascript
const res = await ky.post('some/path', {
  searchParams: {
    'some-param': true
  }
})
```

```
Error: The `searchParams` option must be either a string, `URLSearchParams` instance or an object with string and number values
      at new Ky (node_modules/ky/umd.js:272:13)
      at Function.ky.(anonymous function) [as post] (node_modules/ky/umd.js:501:37)
      ...
```

Is there any reason why boolean values are not supported in the `searchParams` option?

If you create a `URLSearchParams` instance, add the boolean values to that and pass it in as the value of `searchParams` it works as expected, though it's just extra boilerplate.

**Comment by sholladay** [comment]
Yes that is a strange limitation.

I'll try to fix this as soon as PR #180 lands. So, tentatively, it'll most likely be in Ky v0.16.

**Comment by sholladay** [comment]
Working on this now. Very cool to see Ky being used in an IPFS project! :)
